storage/google/cloud/storage/blob.py

```diff
@@ -2057,3 +2064,29 @@ def _add_query_parameters(base_url, name_value_pairs):
     query = parse_qsl(query)
     query.extend(name_value_pairs)
     return urlunsplit((scheme, netloc, path, urlencode(query), frag))
+
+
+class FutureBoolResult(object):
+    """Represents the result of future api request.
+
+    This can be used as a target object for the future request
+    (which is part of batch mechanism) to get result of this request.
+    """
+    _properties = {}
+
+    @property
+    def _is_successfull(self):
+        """Checks if binded future request have been completed without any exception."""
+        return not isinstance(self._properties, _FutureDict)
```

Comment:
If exception have been raise or request is not done yet, there will be `_FutureDict` obj